lib/V8/v8-deadline.cpp

```diff
@@ -0,0 +1,114 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2016 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Wilfried Goesgens
+////////////////////////////////////////////////////////////////////////////////
+#include "V8/v8-deadline.h"
+#include "Basics/system-functions.h"
+#include "v8-utils.h"
+#include "V8/v8-conv.h"
+#include "V8/v8-globals.h"
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief set a point in time after which we will abort external connection
+////////////////////////////////////////////////////////////////////////////////
+static double executionDeadline = 0.0;
+
+// arangosh only: set a deadline
+static void JS_SetExecutionDeatlineTo(v8::FunctionCallbackInfo<v8::Value> const& args) {
+  TRI_V8_TRY_CATCH_BEGIN(isolate);
+  v8::HandleScope scope(isolate);
+
+  // extract arguments
+  if (args.Length() != 1) {
+    TRI_V8_THROW_EXCEPTION_USAGE("SetGlobalExecutionDeadlineTo(<timeout>)");
+  }
+
+  auto when = executionDeadline;
+  auto now = TRI_microtime();
+
+  auto n = TRI_ObjectToUInt64(isolate, args[0], false);
+  if (n == 0) {
+    executionDeadline = 0.0;
+  } else {
+    executionDeadline = TRI_microtime() + n / 1000;
+  }
+
+  TRI_V8_RETURN_BOOL((when > 0.00001) && (now - when > 0.0) );
+  TRI_V8_TRY_CATCH_END
+}
+
+bool isExecutionDeadlineReached(v8::FunctionCallbackInfo<v8::Value> const& args) {
```

Comment:
I was a bit surprised why this function would get `args` from multiple callers without knowing what the args mean.
But it seems like it does not need _args_ but the _isolate_. So we shouldn't pass args, but the isolate to this function.